### PR TITLE
Simplify PDF downloads

### PR DESCRIPTION
**Pull request by geek-soft**
### JIRA link

None

### Change description

This PR is based on https://github.com/hmcts/cmc-citizen-frontend/pull/561 as a further improvement.

Main points:
- DocumentClient returns Buffer that represents array of bytes instead of low level HTTP response
- Download helper is used make return byte array as PDF attachment 

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/app/documents/documentsClient.ts [hunk 1]
```diff
@@ -32,12 +35,14 @@ export class DocumentsClient {
     if (StringUtils.isBlank(bearerToken)) {
       throw new Error('User authorisation cannot be blank')
     }
-    return requestNonPromise.get(
```
Comment (timja-kainos) [inline src/main/app/documents/documentsClient.ts]: is requestNonPromise still used for anything?

Comment (geek-soft) [reply src/main/app/documents/documentsClient.ts]: No, would you like me to remove it?

File: src/main/services/sealedClaimPdfGenerator.ts [hunk 2]
```diff
@@ -16,27 +16,12 @@ export class SealedClaimPdfGenerator {
...
-        })
-        .on('error', (err: Error) => {
-          next(err)
-        })
-    } catch (error) {
-      next(error)
-    }
-
+    const pdf: Buffer = await documentsClient.getSealedClaimPDF(claim.externalId, res.locals.user.bearerToken)
+    DownloadUtils.downloadPDF(res, pdf, `sealed-claim-${claim.claimNumber}`)
```
Comment (geek-soft) [inline src/main/services/sealedClaimPdfGenerator.ts]: Shall I change file name to `{claim.claimNumber}-sealed-claim` for consistency?

File: src/main/app/documents/documentsClient.ts (hunk 1 above)
Comment (timja-kainos) [reply src/main/app/documents/documentsClient.ts]: would make sense to as it was only added for this

File: src/main/services/sealedClaimPdfGenerator.ts (hunk 2 above)
Comment (timja-kainos) [reply src/main/services/sealedClaimPdfGenerator.ts]: i would say so, I honestly don't think defendant's care about `sealed` though

File: src/main/app/documents/documentsClient.ts (hunk 1 above)
Comment (geek-soft) [reply src/main/app/documents/documentsClient.ts]: Will do 👍

Comment (geek-soft) [reply src/main/app/documents/documentsClient.ts]: Done

File: src/main/services/sealedClaimPdfGenerator.ts (hunk 2 above)
Comment (geek-soft) [reply src/main/services/sealedClaimPdfGenerator.ts]: Renamed

**Pull request merged**